lib/reverse_sentence.rb

```diff
@@ -1,6 +1,53 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2), the first `while` loop in `reverse_sentence` will run 'n' times, and each time it calls the `insertion_sort` method that method could run 'n' times. 
+# Space complexity: O(n), where 'n' is the length of `my_sentence`.
+
+
+# helper method to shift each element over once place
+# I believe this is an insertion sort
+def insertion_sort(first_index, my_sentence, last_position, last_character)
+  while first_index < last_position
+    my_sentence[first_index] = my_sentence[first_index + 1]
+    first_index += 1
+  end
+  my_sentence[last_position] = last_character
+end
+
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  if !my_sentence.nil?
+    word_length = 0
+    last_position = my_sentence.length - 1
+    while word_length < last_position
```

Comment:
This loop is shifting every item over one index, then putting the item that was at that index onto the end.  It's doing this for each letter in the string so that's O(n<sup>2</sup>)

You can actually do this in O(n), by having two variables one starting at the front of the string and one at the last letter.  Then swapping the elements at each index.  Then one gets one bigger and the other gets one smaller.  You repeat until they meet in the middle.